README.md

```diff
@@ -56,7 +57,178 @@ Builds the app for production to the build folder.
 
 @TODO: Add production build instructions.
 
-
 ## Event Model
 
 ![Event model](docs/EventModel.png)
+
+## How to create a template (WIP - perhaps move this section to another place)
+
+The templates are created with react ([remote component](https://github.com/Paciolan/remote-component)). The components that are loaded in the client should be compiled.
+TODO: write more about remote component when implemented
+
+### Obligatory
+
+The client loads a compiled react component.
+A component will be created like below
+
+```html
+<ExampleComponent
+  slide={slide}
+  slideDone={slideDone}
+  content={slide.content}
+  run={run}
+/>
+```
+
+**slide**: The slide object.
+**slideDone**: A function that is called when the slide is done.
+**Content**: is the slide content.
+**Run**: is a boolean whether the slide is running or not.
+
+And the following code snippet should be in the component:
```

Comment:
The slide is responsible for signaling that it is done executing. This is done by calling the slideDone() function. If the slide should just run for X milliseconds then you can use the BaseSlideExecution class to handle this. See the example below.